sopel/run_script.py

```diff
@@ -191,23 +123,23 @@ def get_configuration(options):
     This may raise a ``sopel.config.ConfigurationError`` if the file is an
     invalid configuration file.
     """
-    config_name = options.config or 'default'
-    config_path = find_config(DEFAULT_HOMEDIR, config_name)
-
-    if not os.path.isfile(config_path):
+    try:
+        bot_config = utils.load_settings(options)
+    except NotFound as error:
         print(
             "Welcome to Sopel!\n"
             "I can't seem to find the configuration file, "
             "so let's generate it!\n")
 
+        config_path = error.filename
         if not config_path.endswith('.cfg'):
             config_path = config_path + '.cfg'
 
         config_path = _create_config(config_path)
+        # try to reload it now that it's created
+        bot_config = Config(config_path)
```

Comment:
It's very unlikely to happen, and when it'll, it might be because 1) there is a bug on Sopel's config or Sopel's module config; or 2) the file can not be created because of access right.

In both case, let it crash and let `run_script.main` handle it!

Reply:
Fair enough. :)